tools/distpartitioning/utils.py

```diff
@@ -0,0 +1,260 @@
+        Node information as read from xxx_nodes.txt file
+    part_ids : numpy array 
+        array of part_ids indexed by global_nid
+
+    Returns:
+    --------
+    numpy ndarray
+        original node_data, numpy ndarray, with two additional columns
+    """
+    global_nids = np.arange(node_data.shape[0], dtype=np.int64)
+    node_data_aug = np.c_[node_data, global_nids]
+
+    #proc_id = [partitions[x[6]] for x in node_data_aug]
+    proc_ids = part_ids[node_data_aug[:,6]]
+    return np.c_[node_data_aug, proc_ids]
+
+def read_nodes_file(nodes_file):
+    """Utility function to read xxx_nodes.txt file
+    
+    Parameters:
+    -----------
+    nodesfile : string
+        Graph file for nodes in the input graph
+    
+    Returns:
+    --------
+    numpy ndarray
+        node data as in the xxx_nodes.txt file
+    """
+    if nodes_file == "" or nodes_file == None:
+        return None
+
+    # Read the file from here.
+    # Assuming the nodes file is a numpy file
+    # nodes.txt file is of the following format
+    # <node_type> <weight1> <weight2> <weight3> <weight4> <global_type_nid> <attributes>
+    # For the ogb-mag dataset, nodes.txt is of the above format.
+    #
+    nodes_data = np.loadtxt(nodes_file, delimiter=' ', dtype='int64')
+    return nodes_data
+
+def read_edges_file(edge_file):
+    """ Utility function to read xxx_edges.txt file
+
+    Parameters:
+    -----------
+    edge_file : string
+        Graph file for edges in the input graph
+
+    Returns:
+    --------
+    numpy ndarray
+        edge data as in the xxx_edges.txt file
+
+    """
+    if edge_file == "" or edge_file == None:
+        return None
+
+    #Read the file from here.
+    #<global_src_id> <global_dst_id> <type_eid> <etype> <attributes>
+    # global_src_id -- global idx for the source node ... line # in the graph_nodes.txt
+    # global_dst_id -- global idx for the destination id node ... line # in the graph_nodes.txt
+
+    edge_data = np.loadtxt(edge_file , delimiter=' ', dtype = 'int64')
+    return edge_data
+
+def read_node_features_file(nodes_features_file):
+    """Utility function to load tensors from a file
+
+    Parameters:
+    -----------
+    nodes_features_file : string
+        Features file for nodes in the graph
+
+    Returns:
+    --------
+    dictionary
+        mappings between ntype and list of features
+    """
+
+    node_features = dgl.data.utils.load_tensors(nodes_features_file, False)
+    return node_features
+
+def read_edge_features_file(edge_features_file):
+    """ Utility function to load tensors from a file
+
+    Parameters:
+    -----------
+    edge_features_file : string
+        Features file for edges in the graph
+
+    Returns:
+    --------
+    dictionary
+        mappings between etype and list of features
+    """
+    edge_features = dgl.data.utils.load_tensors(edge_features_file, True)
+    return edge_features
+
+def write_node_features(node_features, node_file):
```

Comment:
Is this function used by anywhere else?

Reply:
Yes. this function is used now when we are creating the node features file in each processes.